Sources/SuperwallKit/StoreKit/StoreKitManager.swift

```diff
@@ -0,0 +1,173 @@
+import Foundation
+import Combine
+
+final class StoreKitManager {
+  var productsById: [String: StoreProduct] = [:]
+  private struct ProductProcessingResult {
+    let productIdsToLoad: Set<String>
+    let substituteProductsById: [String: StoreProduct]
+    let products: [Product]
+  }
+
+  private let factory: StoreKitCoordinatorFactory
+  // swiftlint:disable implicitly_unwrapped_optional
+  /// Coordinates: The purchasing, restoring and retrieving of products; the checking
+  /// of transactions; and the determining of the user's subscription status.
+  var coordinator: StoreKitCoordinator!
+  private var receiptManager: ReceiptManager!
+  // swiftlint:enable implicitly_unwrapped_optional
+
+  /// **NOTE:** Remember to call `postInit()` after init
+  init(factory: StoreKitCoordinatorFactory) {
+    self.factory = factory
+  }
+
+  ///  Separated from init to stop circular references during init.
+  func postInit() {
+    coordinator = factory.makeStoreKitCoordinator()
```

Comment:
If I was a betting man, this postInit stuff would've been the part I'd have put good money on you commenting on. I don't like it either. Originally everything in dependency container was lazy so it could just manage everything itself. But then it threw errors of circular references due to refs inside dependency container not being weak (despite the actual refs inside classes being unowned). Then later on I actually had coordinator as lazy and it crashed. But now I just tested and it's fine. Quite a lot has changed in the storekitmanager though since when I first introduced it and can't remember exactly what caused that.
It's worth noting that `lazy` variables aren't thread safe so I am hesitant in using them unless we're sure when the class will be initialised. For example in DeviceHelper a lot of the variables were lazy and that caused a crash for a client who was doing stuff in the background